File: API/Models/ItemTemplate.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using API.Enums;
using Newtonsoft.Json;

namespace API.Models{
    public class ItemTemplate{
        public ItemTemplate(){}
        public ItemTemplate(int id, string name, UnitType unitType, string description, ICollection<TemplatePropertyRelation> properties, ICollection<ItemTemplatePart> parts, string files){
            this.Id = id;
            this.Name = name;
            this.UnitType = unitType;
            this.Description = description;
            this.TemplateProperties = properties;
            this.Parts = parts;
            this.Files = files;
        }

        public ItemTemplate(string name, UnitType unitType, string description, ICollection<TemplatePropertyRelation> properties, ICollection<ItemTemplatePart> parts, string files){
            this.Name = name;
            this.UnitType = unitType;
            this.Description = description;
            this.TemplateProperties = properties;
            this.Parts = parts;
            this.Files = files;
        }
        
        [Key]
        public int Id { get;  set; }
        public string Name { get; set; }
        public UnitType UnitType { get; set; }
        public string Description { get; set; }
        public ICollection<TemplatePropertyRelation> TemplateProperties { get; set; }
        public bool IsActive { get; set; }
        public ICollection<ItemTemplatePart> Parts { get; set; }
        public ICollection<ItemTemplatePart> PartOf { get; set; }
        public string Files { get; set; }

    }
}

File: spa/src/app/_models/ItemTemplate.ts
import {ItemPropertyName} from './ItemPropertyName';
import { ItemTemplatePart } from './ItemTemplatePart';
import { ItemTemplatePartOf} from './ItemTemplatePartOf';
import { DetailFile } from './DetailFile';
import { ItemTemplateCategory } from './ItemTemplateCategory';
import { UnitType } from './UnitType';

export interface ItemTemplate {
    id: number;
    name: string;
    unitType: UnitType;
    revisionedFrom: ItemTemplate;
    created: Date;
    files: any;
    fileNames: string[];
    templateProperties: ItemPropertyName[];
    parts: ItemTemplatePart[];
    partOf: ItemTemplatePartOf[];
    description: string;
    lowerLimit: number;
    category: ItemTemplateCategory;
}
